arch/xtensa/src/esp32/esp32_wifi_adapter.c

```diff
@@ -3447,7 +3473,35 @@ static void *esp_malloc_internal(size_t size)
 
 static void *esp_realloc_internal(void *ptr, size_t size)
 {
-  return kmm_realloc(ptr, size);
+#ifdef CONFIG_XTENSA_IMEM_USE_SEPARATE_HEAP
+  return xtensa_imm_realloc(ptr, size);
+#else
+  void *old_ptr = ptr;
+  void *new_ptr = NULL;
+  size_t old_size = 0;
+  if (size == 0 || esp32_ptr_extram(old_ptr))
```

Comment:
for `size == 0` case, you should free `ptr`.

i don't think you need to check esp32_ptr_extram(old_ptr). the old_ptr will be freed on successful realloc anyway.

Reply:
Done